kkm/kkm_kontext.c

```diff
@@ -411,6 +411,8 @@ int kkm_kontext_switch_kernel(struct kkm_kontext *kkm_kontext)
 	 */
 	local_irq_enable();
 
+	put_cpu();
```

Comment:
After this point the kontext is preemptable.
Have we done something here to ensure the rest of the code (below) is still running on the same cpu?

Reply:
once state is saved there is no reason to be on the same cpu. 
Line 340 makes sure the correct tlb entries exist.